File: estruturas.c
#include <stdlib.h>


typedef struct nob{

     int n;
     char * chaves;
     struct nob * filhos;
}noB;

typedef struct nobl{
     noB * no;
     struct nobl * prox;

}noB_Lista;

typedef struct arvB{

     int ordem;
     noB * raiz;
} TreeB;

File: estruturas.py
class Bno:

     def __init__( self ):
          
          self.n = 0
          self.chaves = []
          self.filhos = []
          self.folha = False

class Btree:

     def __init__( self , ordem ):

          self.ordem = ordem
          self.raiz = Bno()
          self.raiz.folha = True

class BPtree:

     def __init__( self , ordem ):

          self.ordem = ordem
          self.raiz = None

class BPno:

     def __init__( self ):
          
          self.n = 0
          self.chaves = []
          self.filhos = []
          self.folha = False
          self.prox = None
